convert/src/main/kotlin/com/github/zeldigas/text2confl/convert/confluence/ReferenceProvider.kt

```diff
@@ -39,24 +44,41 @@ class ReferenceProviderImpl(private val basePath: Path, documents: Map<Path, Pag
     ReferenceProvider {
 
     companion object {
-        private val URI_DETECTOR = "^[a-zA-Z][a-zA-Z0-9.+-]+:/{0,2}".toRegex()
+        private const val URI_DETECTOR = "^(https?|ftp)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;'()]*[-a-zA-Z0-9+&@#/%=~_|]"
+        private const val MAILTO_DETECTOR = "mailto:"
+        private const val LOCALHOST_DETECTOR = "localhost:"
+
+        private val logger = KotlinLogging.logger {}
+    }
+    fun isValid(url: String): Boolean {
+        val pattern = Pattern.compile(URI_DETECTOR, Pattern.CASE_INSENSITIVE);
```

Comment:
please use regex as a field instead to avoid compiling regex string every time. Original code worked better, so please rollback to original approach.